src/ripple/app/tests/Transaction_ordering_test.cpp

```diff
@@ -30,22 +30,31 @@ struct Transaction_ordering_test : public beast::unit_test::suite
     {
         using namespace std::chrono_literals;
 
+        std::mutex m;
         std::condition_variable cv;
+        bool coroComplete = false;
+
         env.app().getJobQueue().postCoro(
             jtCLIENT, "Coroutine-Test",
             [&](std::shared_ptr<JobCoro> jc)
         {
             env(tx);
+
+            {
+                std::unique_lock<std::mutex> lk(m);
+                coroComplete = true;
+            }
+
             cv.notify_one();
         });
 
         {
-            std::mutex m;
             std::unique_lock<std::mutex> lk(m);
             // If stepping through this test in a debugger,
             // make the timeout much longer, or use
             //cv.wait(lk, test);
-            expect(cv.wait_for(lk, 2s, test));
+            expect(cv.wait_for(lk, 2s,
+                [&]() { return coroComplete && test(); }));
```

Comment:
Here ya go. The `submitWait` function predated `JobQueue::rendezvous`.

https://github.com/ximinez/rippled/commit/c4891d3e3e311dd4c3bb52c8337dff33bf8642d0